bootstrap.sh

```diff
@@ -291,43 +291,5 @@ if [ "$BUILD_PYTHON" == 1 ] ; then
   PYTHONPATH='' $PIP install mysql-connector-python
 fi
 
-#
-# 2. Installation of Go tools and vendored Go dependencies.
-#
-
-
-# Install third-party Go tools used as part of the development workflow.
-#
-# DO NOT ADD LIBRARY DEPENDENCIES HERE. Instead use govendor as described below.
-#
-# Note: We explicitly do not vendor the tools below because a) we want to stay
-# on their latest version and b) it's easier to "go install" them this way.
-gotools=" \
```

Comment:
I think we still need these to be 'gotten', because their binaries have to be built and added to the path. I don't think go modules will do this. These are used from `misc/git/hooks` as pre-commit hooks.

Reply:
> Still need to confirm about gotools.

I am going to look at moving this to a conditional which is shared by the git hooks. That way (assuming java and python are removed as dependencies) `bootstrap.sh` can be run efficiently inside Docker. Currently, it runs before the Docker images are created which causes problems with CI.

I will get to it soon :-)

Reply:
I have added this back, in a conditional with the hooks.

Reply:
https://github.com/golang/go/issues/25922 has some context on how some people deal with tools using in build processes

An option might be to create tools/tools.go something like this and then call `go run github.com/golang/lint/golint` or similar to invoke the tools from within scripts. tools.go would look something like this:
```
// +build tools

package tools

// These imports prevent "go mod tidy" from removing build-time
// dependencies on tools like linters, code generators, etc.
import (
	_ "github.com/golang/lint/golint"
	_ "github.com/golang/mock/mockgen"
	_ "golang.org/x/tools/cmd/cover"
	_ "golang.org/x/tools/cmd/goimports"
	_ "golang.org/x/tools/cmd/goyacc"
	_ "honnef.co/go/tools/cmd/unused"
)
```